cli/executor.cpp

```diff
@@ -45,3 +51,15 @@ bool Executor::hasToLog(const ErrorMessage &msg)
     return false;
 }
 
+void Executor::reportStatus(std::size_t fileindex, std::size_t filecount, std::size_t sizedone, std::size_t sizetotal)
+{
+    if (filecount > 1) {
+        std::ostringstream oss;
+        const long percentDone = (sizetotal > 0) ? lround(static_cast<long double>(sizedone) / sizetotal * 100) : 0;
```

Comment:
This used to be a cast to `long` which caused the value to be wrong after the cast (but somehow not consistently) because of a loss in precision (see some weird handling I introduced temporarily). I am trying to file a ticket about detecting this. The compiler warnings about it are confusing.

Reply:
percentDone is a value from 0-100 so a int would definitely be enough.

Reply:
It looks like cppcheck will say that checking is 100% done when it is only 99.6% done. It's not critical but seems unfortunate to me.

Reply:
I don't understand the old code.. why did it mess with double at all?

why not write `percentDone = (sizetotal > 0) ? (100 * sizedone) / sizetotal : 0;`

Reply:
In the tests with 100 files of the same size you would expect the percentage to go up by 1% each. But it was skipping some steps even though there were no fractions because casting a floating-point to an integer causes precision loss.

The simpler code fixed it. https://godbolt.org/z/W7zo9aT1W


Not using floating-point at all will probably also cause it to jump to 100% too early - if you want t prevent that you probably need to make sure that `99.x` is always rounded down but seems a bit silly - you also still have the file count.

Reply:
> But it was skipping some steps even though there were no fractions because casting a floating-point to an integer causes precision loss.

Such problems are what I expect when fractions are used. And that the output can be different on different machines.

lround will not solve much.. for instance when value is 0.5% then I believe that sometimes the output from lround will be 1 and other times 0.

> Not using floating-point at all will probably also cause it to jump to 100% too early 

integer division rounds down.. (100 * 999)/1000 => 99

Reply:
Great. So this is fixed properly now. I already applied the changes.